src/System.Net.Primitives/src/System/Net/SocketException.Unix.cs

```diff
@@ -25,9 +25,18 @@ private SocketException(Interop.ErrorInfo error) : this(SocketErrorPal.GetSocket
 
         private static int GetNativeErrorForSocketError(SocketError error)
         {
-            return error != SocketError.SocketError ?
-                SocketErrorPal.GetNativeErrorForSocketError(error).Info().RawErrno :
-                (int)error;
+            int nativeErr = (int)error;
+            if (error != SocketError.SocketError)
+            {
+                Interop.Error interopErr = SocketErrorPal.GetNativeErrorForSocketError(error);
+                // If an interopErr was not found, then don't invoke Info().RawErrno as that will fail with assert.
+                if (nativeErr != (int)interopErr)
```

Comment:
Ok, thanks for the clarification.  If you don't end up adding it, at least a comment would be good.